targets/TARGET_STM/TARGET_STM32L4/gpio_irq_api.c

```diff
@@ -304,7 +304,7 @@ void gpio_irq_set(gpio_irq_t *obj, gpio_irq_event event, uint32_t enable)
                 mode = STM_MODE_IT_FALLING;
                 obj->event = EDGE_FALL;
             } else { // NONE or RISE
-                mode = STM_MODE_IT_EVT_RESET;
+                mode = STM_MODE_INPUT;
```

Comment:
This is consistent with the other assert PRs: https://github.com/ARMmbed/mbed-os/pull/3408/files#diff-a565ec22e29e741b7eb21dad81316cc6R307

That GPIO mode is apparently not allowed by the ST HAL